File: admin/src/dtos/City.ts
import Common from "../utility/Common";
import { CityNameRes } from "./CityName";
import { PagedReq } from "./PagedReq";
import { StateRes } from "./State";
import { TimezoneRes } from "./Timezone";

export interface CityRes {
  cityId?: string;
  slug?: string;
  latitude?: number;
  longitude?: number;

  stateId?: string;
  state?: StateRes;

  timezoneId?: string;
  timezone?: TimezoneRes;

  cityNames?: CityNameRes[];
}

export class CityReqEdit {
  slug?: string = "";
  latitude?: number = 0;
  longitude?: number = 0;
  stateId?: string = "";
  timezoneId?: string = "";

  constructor(stateId?: string) {
    this.stateId = stateId;
  }
}

export class CityReqSearch extends PagedReq {
  stateId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {stateId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.stateId = stateId;
  }
}

File: dto/dtos/City.cs
using dto.Paging;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System;

namespace dto.dtos
{
    public class CityRes
    {
        public int CityId { get; set; }
        public string? Slug { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        // Foreign keys
        public int? StateId { get; set; }
        public StateRes? State { get; set; }

        public int? TimezoneId { get; set; }
        public TimezoneRes? Timezone { get; set; }

        // Child tables
        public IEnumerable<CityNameRes>? CityNames { get; set; }
    }
    public class CityReqEdit
    {
        [Required]
        public string? Slug { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }

        // Foreign keys
        public int? StateId { get; set; }
        
        public int? TimezoneId { get; set; }
    }
    public class CityReqSearch : PagedReq
    {
        public int? StateId { get; set; }
    }
}
